libsolidity/inlineasm/AsmAnalysis.cpp

```diff
@@ -447,17 +447,25 @@ void AsmAnalyzer::expectValidType(string const& type, SourceLocation const& _loc
 
 void AsmAnalyzer::warnOnFutureInstruction(solidity::Instruction _instr, SourceLocation const& _location)
 {
+	string instr;
 	switch (_instr)
 	{
+	case solidity::Instruction::CREATE2:
+		instr = "create2";
+		break;
 	case solidity::Instruction::RETURNDATASIZE:
+		instr = "returndatasize";
+		break;
 	case solidity::Instruction::RETURNDATACOPY:
-		m_errorReporter.warning(
-			_location,
-			"The RETURNDATASIZE/RETURNDATACOPY instructions are only available after "
-			"the Metropolis hard fork. Before that they act as an invalid instruction."
-		);
+		instr = "returndatacopy";
 		break;
 	default:
 		break;
 	}
+	if (!instr.empty())
```

Comment:
It would make this much simpler if we would move `instructions()` and `instructionNames()` from `AsmParser`.